File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
N = int(input())

if N % 5 ==0: # N이 5로 나누어 떨어질 때
    print(N // 5)

else: # N이 5로 나누어 떨어지지 않을 때
    sugar = 0
    
    while N > 0:
        N -= 3 # 3KG 짜리 설탕
        sugar += 1

        if N % 5 ==0:
            sugar = sugar + (N//5)
            print(sugar)
            break

        elif N == 1 or N == 2: # 3 또는 5로 안 나누어질 경우
            print('-1')
            break

        elif N == 0: # 3으로만 나누어 떨어질 경우
            print(sugar)
            break

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int sugar = 0;

        if (N % 5 == 0) { // 5로 나누어 떨어질 때
            System.out.println(N / 5);
        } else { // 5로 나누어 떨어지지 않을 때
            while (N>0) {
                N -= 3; // 3짜리 설탕
                sugar += 1;

                if (N % 5 == 0) {
                    sugar += N/5;
                    System.out.println(sugar);
                    break;
                } else if (N == 1 || N == 2) {
                    System.out.println(-1);
                } else if (N == 0) {
                    System.out.println(sugar);
                    break;
                }
            }
        }
    }
}
